mozilla-release/mobile/android/base/java/org/mozilla/gecko/activitystream/homepanel/topnews/TopNewsAdapter.java

```diff
@@ -75,21 +87,59 @@ public void swap(List<TopNews> data) {
         add(data);
     }
 
+    private void toggleNews() {
+        final int changedCount = topNews.size() - COLLAPSED_NUM_NEWS_COUNT;
+        if (isNewsExpanded) {
+            this.numShowedNews = topNews.size();
+            if (changedCount > 0) {
+                notifyItemRangeInserted(COLLAPSED_NUM_NEWS_COUNT + 1, changedCount);
+            }
+        } else {
+            this.numShowedNews = Math.min(topNews.size(), COLLAPSED_NUM_NEWS_COUNT);
+            if (changedCount > 0) {
+                notifyItemRangeRemoved(COLLAPSED_NUM_NEWS_COUNT + 1, changedCount);
+            }
+        }
+        final SharedPreferences.Editor editor = sharedPreferences.edit();
+        editor.putBoolean(GeckoPreferences.PREFS_CLIQZ_TAB_NEWS_EXPANDED, isNewsExpanded);
+        editor.apply();
+    }
+
     @Override
-    public void onBindViewHolder(TopNewsCard holder, int position) {
-        holder.bind(topNews.get(position));
+    public void onBindViewHolder(RecyclerView.ViewHolder holder, int position) {
+        if (holder instanceof TopNewsCard) {
```

Comment:
You can use this function "getItemViewType" to detect the type instead instance of.

Reply:
Did that. Thanks!